src/app/Recordings/ActiveRecordingsTable.tsx

```diff
@@ -313,18 +319,44 @@ export const ActiveRecordingsTable: React.FunctionComponent<ActiveRecordingsTabl
     return () => window.clearInterval(id);
   }, [refreshRecordingList, context, context.settings]);
 
+  const handlePostActions = React.useCallback(
+    (action: ActiveActions) => {
+      setActionLoadings((old) => {
+        const newActionLoadings = { ...old };
+        newActionLoadings[action] = false;
+        return newActionLoadings;
+      });
+    },
+    [setActionLoadings]
+  );
+
   const handleArchiveRecordings = React.useCallback(() => {
+    setActionLoadings((old) => ({ ...old, ARCHIVE: true }));
     const tasks: Observable<boolean>[] = [];
     filteredRecordings.forEach((r: ActiveRecording) => {
       if (checkedIndices.includes(r.id)) {
         handleRowCheck(false, r.id);
         tasks.push(context.api.archiveRecording(r.name).pipe(first()));
       }
     });
-    addSubscription(forkJoin(tasks).subscribe(() => {} /* do nothing */, window.console.error));
-  }, [filteredRecordings, checkedIndices, handleRowCheck, context.api, addSubscription]);
+    addSubscription(
+      forkJoin(tasks).subscribe({
+        next: () => handlePostActions('ARCHIVE'),
+        error: () => handlePostActions('ARCHIVE'),
+      })
+    );
+  }, [
+    filteredRecordings,
+    checkedIndices,
+    handleRowCheck,
+    context.api,
+    addSubscription,
+    setActionLoadings,
+    handlePostActions,
+  ]);
 
   const handleStopRecordings = React.useCallback(() => {
+    setActionLoadings((old) => ({ ...old, STOP: true }));
```

Comment:
is this equivalent to handlePostActions('STOP') ?

Reply:
hmm actually, the `handlePostActions` will reset this to `false` to disable loading.

Reply:
Ah okay.